orm/src/main/java/org/hibernate/search/query/hibernate/impl/FullTextQueryImpl.java

```diff
@@ -289,11 +290,20 @@ public FullTextQuery setResultTransformer(ResultTransformer transformer) {
 		return this;
 	}
 
+	/*
+	 * Implementation note: this method is defined as generic in the interface,
+	 * but we must implement it without generics (otherwise it won't compile).
+	 *
+	 * The actual reason is a bit hard to explain: basically we implement
+	 * javax.persistence.Query as a raw type at some point, and our superclass
+	 * (also extended as a raw type) also implements this interface, but as a non-raw type.
+	 * This seems to confuse the compiler, which thinks there are two different methods.
+	 */
```

Comment:
@Sanne From what I understand of type erasure, my code should be strictly equivalent to the previous code.
I can have a try if you want, compile a small application against 5.6 and then try to run it against 5.7?